agent/agent_test.go

```diff
@@ -3433,6 +3433,20 @@ func TestAgent_ReloadConfigOutgoingRPCConfig(t *testing.T) {
 	require.Len(t, tlsConf.ClientCAs.Subjects(), 2)
 }
 
+func TestAgent_GetRLimits(t *testing.T) {
+	t.Parallel()
+	hcl := `
+		limits{
+			# We put a very high value to be sure to fail
+			# This value is more than max on Windows as well
+			http_max_conns_per_client = 16777217
+		}`
+	_, _, validationError := TestConfigWithErr(testutil.Logger(t), config.Source{Name: t.Name(), Format: "hcl", Data: hcl})
```

Comment:
It can be difficult to understand what is being tested with helpers hide the function-under-test.  `TestConfig` as a healer for testing other code makes a lot of sense, but in this case since we are testing validate I think we want to be calling `Builder.Validate` or `Builder.BuildAndValidate` directly, not through a helper.